browser/src/neovim/NeovimInstance.ts

```diff
@@ -482,27 +497,33 @@ export class NeovimInstance extends EventEmitter implements INeovimInstance {
             // Workaround for bug in neovim/node-client
             // The 'uiAttach' method overrides the new 'nvim_ui_attach' method
             Performance.startMeasure("NeovimInstance.Start.Attach")
-            return this._attachUI(size.cols, size.rows).then(
-                async () => {
-                    Log.info("Attach success")
-                    Performance.endMeasure("NeovimInstance.Start.Attach")
-
-                    // TODO: #702 - Batch these calls via `nvim_call_atomic`
-                    // Override completeopt so Oni works correctly with external popupmenu
-                    // await this.command("set completeopt=longest,menu")
-
-                    // set title after attaching listeners so we can get the initial title
-                    await this.command("set title")
-
-                    Performance.endMeasure("NeovimInstance.Start")
-
-                    this._initComplete = true
-                },
-                (err: any) => {
-                    this._onError(err)
-                    this._initComplete = true
-                },
-            )
+            return this._attachUI(version, size.cols, size.rows)
```

Comment:
@CrossR this definitely pre-dates your change but noted a while ago we had a mixture of patterns here i.e. `async/await` and a `promise chain`, was wondering could the attach ui call not be
```ts
try	{
  await this._attachUI(args)
  if (startUpErrorsChecked === false) {
    this._checkAndFixIfBlocked()
  }
  Log.info("Attach success")
  Performance.endMeasure("NeovimInstance.Start.Attach")
  // TODO: #702 - Batch these calls via `nvim_call_atomic`
  // Override completeopt so Oni works correctly with external popupmenu
  // await this.command("set completeopt=longest,menu")
  // set title after attaching listeners so we can get the initial title
  await this.command("set title")
  Performance.endMeasure("NeovimInstance.Start")
  this._initComplete = true
} catch(error) {
 this._onError(err)
 this._initComplete = true
}
```
Its a lot more seemingly linear and more readable (its a minor point stylistic point 😄)

Reply:
Makes sense!